src/apack/zcl_abapgit_apack_helper.clas.abap

```diff
@@ -0,0 +1,322 @@
+
+    TYPES:
+      BEGIN OF ty_dependency_status,
+        met(1) TYPE c.
+        INCLUDE TYPE zif_abapgit_apack_definitions=>ty_dependency.
+    TYPES: END OF ty_dependency_status,
+      tt_dependency_status TYPE STANDARD TABLE OF ty_dependency_status WITH NON-UNIQUE DEFAULT KEY.
+
+    CLASS-METHODS get_dependencies_met_status
+      IMPORTING
+        !it_dependencies TYPE zif_abapgit_apack_definitions=>tt_dependencies
+      RETURNING
+        VALUE(rt_status) TYPE tt_dependency_status
+      RAISING
+        zcx_abapgit_exception.
+
+    CLASS-METHODS get_installed_packages
+      RETURNING
+        VALUE(rt_packages) TYPE zif_abapgit_apack_definitions=>tt_descriptor
+      RAISING
+        zcx_abapgit_exception.
+
+    CLASS-METHODS show_dependencies_popup
+      IMPORTING
+        !it_dependencies TYPE tt_dependency_status
+      RAISING
+        zcx_abapgit_exception.
+
+ENDCLASS.
+
+
+
+CLASS zcl_abapgit_apack_helper IMPLEMENTATION.
+
+
+  METHOD are_dependencies_met.
+
+    DATA: lt_dependencies_status TYPE tt_dependency_status.
+
+    IF it_dependencies IS INITIAL.
+      rv_status = 'Y'.
+      RETURN.
+    ENDIF.
+
+    lt_dependencies_status = get_dependencies_met_status( it_dependencies ).
+
+    LOOP AT lt_dependencies_status TRANSPORTING NO FIELDS WHERE met <> 'Y'.
+      EXIT.
+    ENDLOOP.
+
+    IF sy-subrc = 0.
+      rv_status = 'N'.
+    ELSE.
+      rv_status = 'Y'.
+    ENDIF.
+
+  ENDMETHOD.
+
+
+  METHOD dependencies_popup.
+
+    DATA: lt_met_status TYPE tt_dependency_status.
+
+    lt_met_status = get_dependencies_met_status( it_dependencies ).
+
+    show_dependencies_popup( lt_met_status ).
+
+  ENDMETHOD.
+
+
+  METHOD get_dependencies_met_status.
+
+    DATA: lt_installed_packages TYPE zif_abapgit_apack_definitions=>tt_descriptor,
+          ls_installed_package  TYPE zif_abapgit_apack_definitions=>ty_descriptor,
+          ls_dependecy          TYPE zif_abapgit_apack_definitions=>ty_dependency,
+          ls_dependecy_popup    TYPE ty_dependency_status.
+
+    IF it_dependencies IS INITIAL.
+      RETURN.
+    ENDIF.
+
+    lt_installed_packages = get_installed_packages( ).
+
+    LOOP AT it_dependencies INTO ls_dependecy.
+      CLEAR: ls_dependecy_popup.
+
+      MOVE-CORRESPONDING ls_dependecy TO ls_dependecy_popup.
+
+      READ TABLE lt_installed_packages INTO ls_installed_package
+        WITH KEY group_id    = ls_dependecy-group_id
+                 artifact_id = ls_dependecy-artifact_id.
+      IF sy-subrc <> 0.
+        ls_dependecy_popup-met = 'N'.
+      ELSE.
+        TRY.
+            zcl_abapgit_version=>check_dependant_version( is_current   = ls_installed_package-sem_version
+                                                          is_dependant = ls_dependecy-sem_version ).
+            ls_dependecy_popup-met = 'Y'.
+          CATCH zcx_abapgit_exception.
+            ls_dependecy_popup-met = 'P'.
```

Comment:
suggest adding constants for these

> met = 'N'.
> met = 'Y'.
> met = 'P'.

Reply:
Ok, should I use those constants also for abapGit requirements Y/N?

Reply:
ideally, yes. But its probably easier to leave all this as-is, both old and new code, and then defer it to a future issue/PR